services/motion/builtin/move_request.go

```diff
@@ -422,7 +432,7 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		}
 	}
 
-	motionCfg, err := newValidatedMotionCfg(req.MotionCfg)
+	motionCfg, err := newValidatedMotionCfg(req.MotionCfg, requestTypeMoveOnGlobe)
```

Comment:
Instead of passing this into the function why not just set it inside the request itself to then be validated by the function?

Reply:
The request itself here is either `motion.MoveOnGlobeReq` or `motion.MoveOnMapReq`. We have not yet created our `move_request` struct so we cannot key off of the requestType that way.

I also do not think we should change the function signature to be accept either a `motion.MoveOnGlobeReq` or `motion.MoveOnMapReq` since, i think, we would then need to use `R[any]` to switch on the types and the default correctly.

Reply:
I dont understand.  The function newMoveOnMapRequest has a `req` that has a motion configuration.  Inside the constructor for that moveRequest we could set this default rather than have to switch on type here

Reply:
ahh, i think I understand.

Reply:
>  Inside the constructor for that moveRequest we could set this default rather than have to switch on type here

I think I mis-understand what you mean.

question: when you say "Inside the constructor for that moveRequest" do you mean `relativeMoveRequestFromAbsolute`?

Within that function we do not know the caller, i.e. we have no way to determine if we are planning for SLAM or for Globe.

Reply:
We could do it right after that call. We already do similar edits at the end of this function:
```
	mr.seedPlan = seedPlan
	mr.replanCostFactor = valExtra.replanCostFactor
	mr.requestType = requestTypeMoveOnGlobe
```
We can set this here.

Reply:
> We could do it right after that call.

We would first need to see if 
```
	if motionCfg.PlanDeviationMM != 0 {
		vmc.planDeviationMM = motionCfg.PlanDeviationMM
	}
```
which is already done in `newValidatedMotionCfg`, so we would be  duplicating code in `newMoveOnGlobeReq` and `NewMoveOnMapReq`

Further, the check to see if we are already at the goal or not is done within the function that construct the `move_request`, i.e. `relativeMoveRequestFromAbsolute`. 

If we set the defaults in a different place, the code to do the check would need to also be duplicating within `newMoveOnGlobeReq` and `NewMoveOnMapReq`.

Reply:
It does seem like we'd be repeating ourselves somewhere otherwise, so I'm good on this.  But if Peter has a better idea would be down with that as well